### PR TITLE
Plugin catkin: disable parallel option

### PR DESCRIPTION
**Pull request by artivis**
- [x] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] Have you successfully run `./runtests.sh static`?
- [ ] Have you successfully run `./runtests.sh tests/unit`?

-----

Catkin plugin : add an option to disable parallel building.

Note: running tests using `spread`, some bits are failing. However the same bits are failing on `master` branch as well.

### CODE REVIEW

**Comment by artivis** [comment]
@kyrofa FYI

File: snapcraft/plugins/catkin.py [hunk 1]
```diff
@@ -68,6 +68,10 @@
       (string)
       The URI to ros master setting the env variable ROS_MASTER_URI. Defaults
       to http://localhost:11311.
+    - disable-parallel:
```
Comment (kyrofa) [inline snapcraft/plugins/catkin.py]: This functionality actually already exists, just use [this](https://github.com/snapcore/snapcraft/blob/master/snapcraft/_baseplugin.py#L174) to enable it. The [make](https://github.com/snapcore/snapcraft/blob/master/snapcraft/plugins/make.py#L108) plugin is a great example.

Comment (artivis) [reply snapcraft/plugins/catkin.py]: I see. I'll let CI finish the current run then open another PR to replace this.

Comment (kyrofa) [reply snapcraft/plugins/catkin.py]: No need to close the PR, just revamp it a bit. Make sure you add a test that verifies that specifying the option actually does result in `-j1`.

Comment (artivis) [reply snapcraft/plugins/catkin.py]: Well, the scope of the feature slightly change and most of the introduced code goes down the drain. I think opening a new PR is proper.  
Also, any tip on testing ?

**Pull request closed without merging**